bokehjs/src/lib/models/annotations/color_bar.ts

```diff
@@ -103,153 +94,68 @@ export class ColorBarView extends AnnotationView {
     Note: the type of color_mapper has to match the type of scale (i.e.
     a LinearColorMapper will require a corresponding LinearScale instance).
     */
-    this._major_range = (() => {
-      if (color_mapper instanceof CategoricalColorMapper) {
-        const {factors} = color_mapper
+    const {color_mapper} = this.model
+
+    switch (true) {
+      case color_mapper instanceof CategoricalColorMapper:
+        const {factors} = color_mapper as CategoricalColorMapper
         return new FactorRange({factors})
-      } else if (color_mapper instanceof ContinuousColorMapper) {
-        const {min, max} = color_mapper.metrics
+      case color_mapper instanceof ContinuousColorMapper:
+        const {min, max} = (color_mapper as ContinuousColorMapper).metrics
         return new Range1d({start: min, end: max})
-      } else
+      default:
         unreachable()
-    })()
+    }
+  }
+
+  override _create_major_scale(): Scale {
+    const {color_mapper} = this.model
 
-    this._major_scale = (() => {
-      if (color_mapper instanceof LinearColorMapper)
+    switch (true) {
```

Comment:
personally I find this construction obscure. Is there a tangible benefit over if/else cascade?

Reply:
No idea! This is code that was already in `color_bar.ts` before I refactored it into `base_color_bar.ts` and `color_bar.ts`. I don't like it either! I can change it whilst I am here if you'd like.

Reply:
I have changed all the `switch (true` JS blocks to `if`-`else` as they seem clearer to me as well.

Reply:
In this case it's a matter of taste more than anything else. However, if you have no idea, then why change it? Especially that it's easy to miss this in the overall shuffling of code around The right thing to do, was to keep the original and start a discussion. Then we would be able to decide what's the best approach.

Reply:
@mattpap It's not merely style preference, it's about keeping the codebase accessible to many, and avoiding extremely obscure and unusual features unless there is a concrete justification to prefer them.